src/cluster/distribution.rst

```diff
@@ -0,0 +1,111 @@
+.. Licensed under the Apache License, Version 2.0 (the "License"); you may not
+.. use this file except in compliance with the License. You may obtain a copy of
+.. the License at
+..
+..   http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+.. WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+.. License for the specific language governing permissions and limitations under
+.. the License.
+
+.. _cluster/distribution:
+
+==============================
+Custom Erlang Network Protocol
+==============================
+The main purpose is to allow users to specify network protocols for CouchDB
+nodes, which enhanced data security during data migration between nodes.
+
+This section will explain how to enable TCP/TLS distribution for additional
+verification and security.
+
+Generate Certificate
+====================================
+For TLS to work properly, at least one public key and one certificate must be
+specified. In the following example (couch_ssl_dist.conf), the PEM file contains
+the ``certificate`` and its ``private key``.
+
+    .. code-block:: text
+
+        [{server,
+          [{certfile, "</path/to/erlserver.pem>"},
+           {secure_renegotiate, true}]},
+         {client,
+          [{secure_renegotiate, true}]}].
+
+The following command is an example of generating a certificate (PEM) file.
+
+    .. code-block:: bash
+
+        $ openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout key.pem -out cert.pem
+        $ cat key.pem cert.pem > erlserver.pem
+        $ rm key.pem cert.pem
+
+    .. note::
+       This is not an endorsement of a specific expiration limit, key size or algorithm.
+
+Config Settings
+====================================
+To enable TLS distribution, make sure to set custom parameters in ``vm.args``.
+
+    .. code-block:: text
+
+        # Don't forget to override the paths to point to your cert and conf file!
+        -proto_dist couch
+        -couch_dist no_tls \"clouseau@127.0.0.1\"
+        -ssl_dist_optfile <path/to/couch_ssl_dist.conf>
+
+    .. note::
+       To ensure "search" works, make sure to set ``no_tls`` option for the
+       ``clouseau`` node. By default, this will be "clouseau@127.0.0.1".
+
+CouchDB recommends the following values for the ``no_tls`` flag:
```

Comment:
I think the only value we'd recommend is to let `clouseau` connect with TCP when using search, otherwise it won't work. Then we'd just say perhaps that  ``` The `no_tls` setting can have these values: ```

Reply:
changed